modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -199,6 +199,21 @@ def _axes_lengths(self):
         """
         return [self.row_lengths, self.column_widths]
 
+    def _set_axis_widths_cache(self, value, axis=0):
```

Comment:
Should we have `set_row_lengths_cache` and `set_column_widths_cache` instead?

Reply:
We tend to name partition sizes as lengths, but not widths so I would suggest to rename the method to `_set_axis_lengths_cache`. Also, do you think if it should be public?

Reply:
Renamed.

I believe the method has to be private as partition's shapes are really an internal thing and we should disclose it.